File: src/leetcode/python/RomanToInteger_13.py
class RomanToInteger_13(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        m = {'M': 1000, 'D': 500, 'C': 100, 'L': 50, 'X': 10, 'V': 5, 'I': 1}
        res = 0
        for i in range(len(s) - 1):
            if m[s[i]] < m[s[i + 1]]:
                res -= m[s[i]]
            else:
                res += m[s[i]]
        return res + m[s[-1]]

File: src/leetcode/java/RomanToInteger_13.java
package leetcode.java;

public class RomanToInteger_13 {
  public int romanToInt(String s) {
    int res = 0;
    int i;
    for (i = 0; i < s.length() - 1; i++) {
      if (roman(s.charAt(i)) < roman(s.charAt(i + 1))) {
        res -= roman(s.charAt(i));
      } else {
        res += roman(s.charAt(i));
      }
    }
    return res + roman(s.charAt(i));
  }

  private static int roman(char c) {
    switch (c) {
      case 'M': return 1000;
      case 'D': return 500;
      case 'C': return 100;
      case 'L': return 50;
      case 'X': return 10;
      case 'V': return 5;
      default: return 1;
    }
  }
}
